csharp/examples/FlightAspServerExample/Services/InMemoryFlightServer.cs

```diff
@@ -0,0 +1,158 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Apache.Arrow;
+using Apache.Arrow.Flight;
+using Apache.Arrow.Flight.Server;
+using Grpc.Core;
+
+namespace FlightAspServerExample.Services
+{
+    public class FlightData
+    {
+        public FlightData()
+        {
+            flights = new Dictionary<FlightTicket, FlightInfo> { };
+            tables = new Dictionary<FlightTicket, List<RecordBatch>> { };
```

Comment:
```suggestion
            flights = new ConcurrentDictionary<FlightTicket, FlightInfo> { };
            tables = new ConcurrentDictionary<FlightTicket, List<RecordBatch>> { };
```

Since `FlightData` is a Singleton service, it needs to be thread-safe.